### DISCUSSION TITLE
Mirth heap space monitor

### DISCUSSION THREAD

**Discussion by jbyrd-mhd** [nextgenhealthcare/connect#4788] [Q&A]
Is there a way to monitor the amount of heap space being used by Mirth? Our plan of the attack in the past has typically been max out heap space, keep adding more channels to process greater amounts of messages until we hit failure, then create a new server and start moving channels over.  Need to find a better way to know that we'll need an additional server soon.

**Comment by pacmano1** [comment]
Can you share a bit about your message volumes, number of channels, average message size?   And server / vmoption configs?    Just a good reference point for others.

**Comment by jbyrd-mhd** [comment]
**Mirth**-
Server version 3.7.1
 
Average message volume: 333k daily with the majority being received during business hours
These 333k messages are received on 8 separate TCP listener source channels and then sent to subchannels based on load distributing logic.  

Config: 2056 mb heap space

**VM**- 
Standard D16s v3 (16 vcpus, 64 GiB memory)

**Reply by pacmano1** [reply]
Intersting.   I'd expect this (the need to add a new instance) at millions of messages per day and that VM is gigantic for that volume of mesages if mostly doing hl7 transforms / write to db and so on.    

You have 64Gb of memory on the server and given mirth a 2G heap space?    Really puzzled (although even at that message volume should be a non-issue).

**Reply by jbyrd-mhd** [reply]
We are thinking the same thing.  What is a reasonable setting for mirth heap space? Are there any repercussions with increasing it quite a bit? 

Our problems seems to reveal themselves more when large queues build across the board.

**Reply by pacmano1** [reply]
All depends on the actual transformations you are doing.  For example, we consume CDAs with Nexgen's CDAPI library which is a bit memory hungry.   The kind of transformations you are doing likely affert needed jvm size.   If you are just routing stuff around and maybe writing message data to a database that is usually low memory.   I'll defer to others on repercusssions of a large jvm.  All sort of considerations including garbage collection and other jvm params.
